helm/main.go

```diff
@@ -0,0 +1,102 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package main
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"text/template"
+)
+
+const (
+	envoyGatewayChartsDirectory = "charts/gateway-helm"
+	templateFileExtension       = ".tmpl"
+)
+
+var (
+	overlays = struct {
+		ImageTag        string
+		ImageRepository string
```

Comment:
I prefer `EGImageTag`, `EGImageRepository`, we do not just have image for eg, and may have other images. I think we should have the prefix to prevent conflicts.